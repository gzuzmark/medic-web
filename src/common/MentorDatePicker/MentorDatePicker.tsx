import * as moment from "moment";
import * as React from "react";
import DatePickerBase from "../DatePickerBase/DatePickerBase";

interface IStateMentorDatePicker {
    focus: boolean;
}

interface IPropsMentorDatePicker {
    id: string;
    configDate?: object;
    date: Date;
    error?: boolean;
    updateState: (params: object) => void;
    label: string;
    attrs: any;
}

// TODO: Se necesita actualizar estilos
class MentorDatePicker extends React.Component <IPropsMentorDatePicker, IStateMentorDatePicker> {
    public state: IStateMentorDatePicker;
    constructor(props: IPropsMentorDatePicker) {
        super(props);
        this.state = {
            focus: false
        };
        this.onDateChange = this.onDateChange.bind(this);
        this.onDateFocusChange = this.onDateFocusChange.bind(this);
    }

    public render() {
        return (
            <div>
                <DatePickerBase
                    id={this.props.id}
                    date={moment(this.props.date)}
                    focus={this.state.focus}
                    onDateChange={this.onDateChange}
                    onDateFocusChange={this.onDateFocusChange}
                    configs={this.props.configDate} />
            </div>
        )
    }

    private onDateChange(date: moment.Moment) {
        this.props.updateState({[this.props.id]: date.toDate()})

    }

    private onDateFocusChange(focusedInput: any) {
        const focus = focusedInput ? focusedInput.focused : false;
        this.setState({focus});
    }


}

export default MentorDatePicker;
