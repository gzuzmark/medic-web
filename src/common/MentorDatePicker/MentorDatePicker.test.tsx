import { shallow } from "enzyme";
import * as React from 'react';
import MentorDatePicker from "./MentorDatePicker";

describe('MentorDatePicker Test',() => {
    let props: any;
    let mountedMentorDatePicker: any;
    const getComponent = () => {
        if (!mountedMentorDatePicker) {
            mountedMentorDatePicker = shallow(
                <MentorDatePicker {...props} />
            );
        }
        return mountedMentorDatePicker;
    };

    beforeEach(() => {
        props = {
            date: new Date("2018-02-02"),
            id: 'demoKey',
            updateState: (params: object) => void(0)
        };
        mountedMentorDatePicker = undefined;
    });

    it("render: render MentorDatePicker", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });
});
