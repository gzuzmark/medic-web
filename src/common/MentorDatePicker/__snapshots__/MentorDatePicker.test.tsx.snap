// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentorDatePicker Test render: render MentorDatePicker 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MentorDatePicker
    date={2018-02-02T00:00:00.000Z}
    id="demoKey"
    updateState={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <DatePickerBase
        configs={undefined}
        date={"2018-02-02T00:00:00.000Z"}
        focus={false}
        id="demoKey"
        onDateChange={[Function]}
        onDateFocusChange={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "configs": undefined,
        "date": "2018-02-02T00:00:00.000Z",
        "focus": false,
        "id": "demoKey",
        "onDateChange": [Function],
        "onDateFocusChange": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <DatePickerBase
          configs={undefined}
          date={"2018-02-02T00:00:00.000Z"}
          focus={false}
          id="demoKey"
          onDateChange={[Function]}
          onDateFocusChange={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "configs": undefined,
          "date": "2018-02-02T00:00:00.000Z",
          "focus": false,
          "id": "demoKey",
          "onDateChange": [Function],
          "onDateFocusChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
